tests/integration/fixtures.py

```diff
@@ -0,0 +1,196 @@
+    assert create_subscriber_resp.status_code == 201
+    create_device_group_resp = _create_device_group(webui_ip_address)
+    assert create_device_group_resp.status_code == 200
+    create_network_slice_resp = _create_network_slice(webui_ip_address)
+    assert create_network_slice_resp.status_code == 200
+
+
+@pytest.fixture(scope="module")
+@pytest.mark.abort_on_fail
+async def deploy_gnbsim(ops_test: OpsTest):
+    """Deploys `sdcore-gnbsim`.
+
+    Args:
+        ops_test: OpsTest
+    """
+    await ops_test.model.deploy(  # type: ignore[union-attr]
+        "sdcore-gnbsim",
+        application_name="gnbsim",
+        channel="latest/edge",
+        trust=True,
+    )
+    await ops_test.model.add_relation("gnbsim:fiveg-n2", "amf:fiveg-n2")  # type: ignore[union-attr]  # noqa: E501
+    await ops_test.model.wait_for_idle(  # type: ignore[union-attr]
+        apps=["gnbsim"],
+        raise_on_error=False,
+        status="active",
+        timeout=300,
+    )
+
+
+async def deploy_cos_lite(ops_test: OpsTest):
+    """Deploys `cos-lite` bundle.
+
+    Args:
+        ops_test: OpsTest
+    """
+    await ops_test.track_model(
+        alias=COS_MODEL_NAME,
+        model_name=COS_MODEL_NAME,
+        cloud_name=ops_test.cloud_name,
+    )
+
+    with ops_test.model_context(COS_MODEL_NAME):
+        await ops_test.model.deploy(  # type: ignore[union-attr]
+            entity_url="https://charmhub.io/cos-lite",
+            trust=True,
+        )
+        await ops_test.model.wait_for_idle(  # type: ignore[union-attr]
+            apps=[*ops_test.model.applications],  # type: ignore[union-attr]
+            raise_on_error=False,
+            status="active",
+            timeout=600,
+        )
+        await ops_test.model.create_offer(  # type: ignore[union-attr]
+            endpoint="prometheus:receive-remote-write",
+            offer_name="prometheus",
+            application_name="prometheus",
+        )
+
+
+async def deploy_cos_configuration(ops_test: OpsTest):
+    """Deploys `cos-configuration-k8s` charm.
+
+    Args:
+        ops_test: OpsTest
+    """
+    with ops_test.model_context(COS_MODEL_NAME):
+        await ops_test.model.deploy(  # type: ignore[union-attr]
+            "cos-configuration-k8s",
+            config={
+                "git_repo": "https://github.com/canonical/sdcore-cos-configuration",
+                "git_branch": "main",
+                "git_depth": 1,
+                "grafana_dashboards_path": "grafana_dashboards/sdcore/",
+            },
+        )
+        await ops_test.model.add_relation("cos-configuration-k8s", "grafana")  # type: ignore[union-attr]  # noqa: E501
+        await ops_test.model.wait_for_idle(  # type: ignore[union-attr]
+            apps=["cos-configuration-k8s"],
+            raise_on_error=False,
+            status="active",
+            timeout=300,
+        )
+
+
+def _create_subscriber(webui_ip: str) -> requests.Response:
```

Comment:
API calls to webui can be in a class in a separate file.

```python
class Webui:
    def __init__(self, url: str):
        self.url = url
    def create_subscriber(self, imsi: str):
        ...
```

Reply:
Done.